GameServer/ai/brain/StandardMobBrain.cs

```diff
@@ -373,17 +370,7 @@ public virtual int AggroRange
 		/// Checks whether living has someone on its aggrolist
 		/// </summary>
 		public virtual bool HasAggro
-		{
-			get
-			{
-				bool hasAggro = false;
-				lock ((m_aggroTable as ICollection).SyncRoot)
-				{
-					hasAggro = (m_aggroTable.Count > 0);
-				}
-				return hasAggro;
-			}
-		}
+			=> AggroTable.Count > 0;

```

Comment:
You are removing a lock here. This gets dangerous for concurrent access.

Reply:
Access to `.Count` is thread-safe as it is calling members of `Dictionary`. Generally speaking `.Count` is meant to be thread-safe/atomic while `.Count()` does not have this promise. Granted the value might mismatch after it has been called, but the lock is not protecting against that.